sklearn/metrics/pairwise.py

```diff
@@ -30,6 +30,8 @@
 from ..utils._joblib import effective_n_jobs
 
 from .pairwise_fast import _chi2_kernel_fast, _sparse_manhattan
+from .pairwise_fast import _euclidean_dense_dense_exact
+from ._safe_euclidean_sparse import _euclidean_sparse_dense_exact
```

Comment:
It might be a bit surprising that these implementations are in different modules. Why did you want them separated?

Reply:
I don't want them to be separated. I had to separate them because I wanted to compile `pairwise_fast` with the `-ffast-math` flag. The thing is that with this flag, gcc performs unsafe optimizations for the `_safe_euclidean_sparse` functions.

The reason I wanted to use this flag for `pairwise_fast` is that it's much faster, and should be safe because there's no operation that gcc would optimize in an unsafe way. By the way, scipy uses this flag for its euclidean distance.